packages/server-core/src/projects/project/project.service.ts

```diff
@@ -36,6 +37,14 @@ export default (app: Application): void => {
     }
   })
 
+  app.use('project-invalidate', {
+    patch: async ({ projectName }, params) => {
+      if (projectName) {
+        return await useStorageProvider().createInvalidation([`projects/${projectName}*`])
+      }
+    }
+  })
+
   app.service('project-build').hooks({
```

Comment:
There should be a .hooks for the new project-invalidate service. It should also add authenticate() and restrictUserRole('admin') on the patch action.

Reply:
yes, really we should move all these to be methods inside of the 'project' patch action. i made a bad pattern here. let's recitfy it here.